restart_process/Tiltfile

```diff
@@ -0,0 +1,76 @@
+RESTART_FILE = '/.restart-proc'
+TYPE_RESTART_CONTAINER_STEP = 'live_update_restart_container_step'
+
+def docker_build_with_restart(ref, context, entrypoint, live_update,
+                              base_suffix='-base', restart_file=RESTART_FILE, **kwargs):
+    """Wrap a docker_build call and its associated live_update steps so that the last step
+    of any live update is to rerun the given entrypoint.
+
+
+     Args:
+      ref: name for this image (e.g. 'myproj/backend' or 'myregistry/myproj/backend'); as the parameter of the same name in docker_build
+      context: path to use as the Docker build context; as the parameter of the same name in docker_build
+      entrypoint: the command to be (re-)executed when the container starts or when a live_update is run
+      live_update: set of steps for updating a running container; as the parameter of the same name in docker_build
+      base_suffix: suffix for naming the base image, applied as {ref}{base_suffix}
+      restart_file: file that Tilt will `touch` during a live_update to signal the entrypoint to rerun
+      **kwargs: will be passed to the underlying `docker_build` call
+    """
+
+    # first, validate the given live_update steps
+    if len(live_update) == 0:
+        fail("`docker_build_with_restart` requires at least one live_update step")
+    for step in live_update:
+        if type(step) == TYPE_RESTART_CONTAINER_STEP:
+            fail("`docker_build_with_restart` is not compatible with live_update step: "+
+                 "`restart_container()` (this extension is meant to REPLACE restart_container() )")
+
+    # rename the original image to make it a base image and declare a docker_build for it
+    base_ref = '{}{}'.format(ref, base_suffix)
+    docker_build(base_ref, context, **kwargs)
+
+    # declare a new docker build that adds a static binary of entr (previously
+    # compiled by tilt) to the user's image
+    df = '''
+    FROM tiltdev/entr:2020-16-04 as entr-img
+
+    FROM {}
+    RUN ["touch", "{}"]
+    COPY --from=entr-img /entr /
+  '''.format(base_ref, restart_file)
+
+    clean_kwargs(kwargs)
+
+    # Change the entrypoint to use entr.
+    # entr allows you to run commands when files change: https://github.com/eradman/entr/
+    # this invocation says: whenever $restart_file changes, re-execute $entrypoint
+    entrypoint_with_entr = "echo '{}' | /entr -rz {}".format(restart_file, entrypoint)
+
+    # last live_update step should always be to modify $restart_file, which
+    # triggers entr to rerun $entrypoint
+    live_update.append(run('date > {}'.format(restart_file)))
```

Comment:
i wouldn't expect this function to mutate its arguments. Probably better as:
live_update = live_update + ...